Tests/Fuzzers/ReadParsedPacket.h

```diff
@@ -0,0 +1,522 @@
+#ifndef READ_PARSED_PACKET_H
+#define READ_PARSED_PACKET_H
+
+#include <Packet.h>
+#include <IPv4Layer.h>
+#include <TelnetLayer.h>
+#include <BgpLayer.h>
+#include <DhcpLayer.h>
+#include <DhcpV6Layer.h>
+#include <DnsLayer.h>
+#include <IcmpLayer.h>
+#include <NtpLayer.h>
+#include <SSLLayer.h>
+#include <SSLHandshake.h>
+#include <TcpLayer.h>
+#include <SdpLayer.h>
+#include <VrrpLayer.h>
+#include <Sll2Layer.h>
+#include <FtpLayer.h>
+#include <GreLayer.h>
+#include <GtpLayer.h>
+#include <SomeIpSdLayer.h>
+
+// Call some pcpp::Packet methods that are not invoked from general virtual methods
+// as `pcpp::Packet::toString` or `pcpp::Packet::computeCalculateFields` to trigger possible crashes.
+// The general rule is the functions do not modify the `parsedPacket`.
+// If you made changes to PcapPlusPlus and the code doesn't compile - fix the method call as any other unit test
+static void readParsedPacket(pcpp::Packet parsedPacket, pcpp::Layer* layer)
```

Comment:
This file covers ~20 protocols out of ~45 supported in PcapPlusPlus. My questions are:
- How did you choose which protocols to include and which not to include?
- When we add a new protocol (we recently added [S7comm](https://github.com/seladb/PcapPlusPlus/pull/1185)), how should we, as maintainers / PR reviewers, guide contributors on whether to add their new protocol to the fuzzing tests or not? Or maybe contributors shouldn't worry about it and we (maintainers) will add it after the fact?
- If a contributor wants to add code to the fuzzing tests, how do they test fuzzing to make sure they don't break anything? Maybe we should add some documentation on how to contribute to this part of the code. So far we didn't need it because the fuzzing tests were simple and very static (we made very little changes in them over the past few years)

Reply:
The current coverage report for pcapplusplus is [here](https://storage.googleapis.com/oss-fuzz-coverage/pcapplusplus/reports/20231106/linux/report.html). It is also possible to generate it [locally according to instructions](https://google.github.io/oss-fuzz/advanced-topics/code-coverage/) if, say, you have local changes.

I was looking into the report seeking functions that are marked as uncovered. For example
`getEventgroupId` - https://storage.googleapis.com/oss-fuzz-coverage/pcapplusplus/reports/20231106/linux/src/PcapPlusPlus/Packet++/src/SomeIpSdLayer.cpp.html#L378
Then I was looking why it is not called. What are the other functions that call it?
As you can see [here for example](https://github.com/search?q=repo%3Aseladb%2FPcapPlusPlus%20getEventgroupId&type=code) the function is called only explicitly from a unit test.

> When we add a new protocol (we recently added https://github.com/seladb/PcapPlusPlus/pull/1185), how should we, as maintainers / PR reviewers, guide contributors on whether to add their new protocol to the fuzzing tests or not? Or maybe contributors shouldn't worry about it and we (maintainers) will add it after the fact?

New protocol contributors should not worry about adding anything to the harness. After some time a new coverage check can be done and the functions added.

> If a contributor wants to add code to the fuzzing tests, how do they test fuzzing to make sure they don't break anything?

Fuzzers are built and run in CI. Oss-fuzz also has [instructions](https://google.github.io/oss-fuzz/getting-started/new-project-guide/#testing-locally) how to build and test locally.

Reply:
This is very useful information! I think we should add a README with this info and useful instructions. I'd include:
- Some general information on what are fuzzing tests
- What fuzzing tests are included here for PcapPlusPlus
- Useful links: 
  - oss-fuzz project
  - The latest coverage report
  - Open issues for PcapPlusPlus
  - How to build locally
  - How to create a coverage report locally
- Anything else you can think of?

Reply:
I have added the readme.